### PR TITLE
tests-libc: add tests of pid/uid related functions in unistd.h

### PR DESCRIPTION
**Pull request by niewim19**
JIRA: CI-100

## Description

 - Adding tests of functions from unistd.h that cope with processes `pid`s and `uid`s
 - Couple IGNOREs as couple issues were found.

## Motivation and Context

Issues related to these tests:
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/279
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/280
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/282

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libc/unistd_uids.c [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+{
+	child = NULL;
+	pid = parent[0] = parent[1] = parent[2] = -1;
+}
+
+
+TEST_TEAR_DOWN(unistd_uids)
+{
+	if (child != NULL)
+		free(child);
```
Comment (agkaminski) [inline libc/unistd_uids.c:39]: free(NULL) is safe
```suggestion
	free(child);
```

File: libc/unistd_uids.c [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+#include <string.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <stdlib.h>
+
+#include <unity_fixture.h>
+
+
+pid_t *child, parent[3], pid;
```
Comment (agkaminski) [inline libc/unistd_uids.c:24]: Only use-case I see is `child = malloc(sizeof(pid_t) * 6);` anyway
```suggestion
pid_t child[6], parent[3], pid;
```

File: libc/unistd_uids.c [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	TEST_ASSERT_NOT_EQUAL_INT(child[3], child[5]);  /* pid != sid */
+}
+
+TEST_GROUP_RUNNER(unistd_uids)
+{
+	RUN_TEST_CASE(unistd_uids, getuids_parent);
+	RUN_TEST_CASE(unistd_uids, setuids_parent);
+	RUN_TEST_CASE(unistd_uids, setpuids_setsid);
+	RUN_TEST_CASE(unistd_uids, setpuids_setpgid)
+}
```
Comment (agkaminski) [inline libc/unistd_uids.c:193]: No new line at the of the file

File: libc/unistd_uids.c [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+TEST(unistd_uids, setpuids_setsid)
+{
+	int err, siderr = -1;
+
+	parent[0] = getpid();
+	parent[1] = getpgrp();
+	parent[2] = getsid(getpid());
+
+	child = malloc(sizeof(pid_t) * 6);
```
Comment (agkaminski) [inline libc/unistd_uids.c:88]:
```suggestion
```

File: libc/unistd_uids.c [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+	*/
+	TEST_IGNORE();
+
+	int err, pgrperr = -1;
+
+	parent[0] = getpid();
+	parent[1] = getpgrp();
+	parent[2] = getsid(getpid());
+
+	child = malloc(sizeof(pid_t) * 6);
```
Comment (agkaminski) [inline libc/unistd_uids.c:144]:
```suggestion
```

File: libc/unistd_uids.c (hunk 4 above)
Comment (niewim19) [reply libc/unistd_uids.c]: Yes. I am correcting that. Also `free(child)` is to be removed. Thanks!

**Review by agkaminski — APPROVED** [review]

**Pull request merged**